### DISCUSSION TITLE
Accessing a singleton instance of #[qml_singleton] from Rust side

### DISCUSSION THREAD

**Discussion by Billyzou0741326** [KDAB/cxx-qt#655] [Ideas]
As per my understanding, `qobject`s marked with `#[qml_singleton]` will be generated as `QML_SINGLETON` on the C++ side.

Is there a way to access an instance of said singleton on the Rust side?

Looks like `QQmlEngine` exposes these methods on the C++ side (https://doc.qt.io/qt-6/qqmlengine.html#singletonInstance):

```c++
T singletonInstance(int qmlTypeId);
T singletonInstance(QAnyStringView uri, QAnyStringView typeName);
```

Is there any existing method to access a singleton instance that I'm not aware of? If not, are there plans to support this use case?

**Comment by Be-ing** [comment]
> `T singletonInstance(QAnyStringView uri, QAnyStringView typeName);`

That looks like the most appropriate way to do it from Rust. We don't have bindings for QAnyStringView yet: https://github.com/KDAB/cxx-qt/issues/629

**Reply by ahayzen-kdab** [reply]
And the templatedness of that makes it fun to do in CXX :-) Developers might need to create a specific binding in their own bridge :thinking: We would need to think about this.
